main.tf

```diff
@@ -128,6 +128,7 @@ data "template_file" "gitlab_runner" {
     gitlab_runner_locked_to_project         = var.gitlab_runner_registration_config["locked_to_project"]
     gitlab_runner_run_untagged              = var.gitlab_runner_registration_config["run_untagged"]
     gitlab_runner_maximum_timeout           = var.gitlab_runner_registration_config["maximum_timeout"]
+    gitlab_runner_access_level              = var.gitlab_runner_registration_config["access_level"]
```

Comment:
We could change this line as follow so adding the property to the map become optional and the change will remain in that case backwards compatible

```
gitlab_runner_access_level              = lookup(var.gitlab_runner_registration_config, "access_level", "not_protected")
```

Reply:
Awesome, added! Great reason why I should look through the TF functions that are available.